### PR TITLE
Add sidebar

### LINKED ISSUE #48: sidebar implementation/fixes

### PR DESCRIPTION
**Pull request by nkelava**
This PR will:
- Adds sidebar feature to existing pages
- Move logout to sidebar
- Move timetable generator view

Fixes #48

### CODE REVIEW

File: frontend/src/components/SideBar/SideBarLink.vue [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<script setup>
+import { computed } from 'vue';
+const props = defineProps({
```
Comment (bdeak4) [inline frontend/src/components/SideBar/SideBarLink.vue:4]: nitpick: please add some space around this variable declaration for better visibility

File: frontend/src/components/SideBar/SideBarLink.vue [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+<style scoped>
+.sidebar-item {
+  display: flex;
+  align-items: center;
+  text-decoration: none;
+  transition: 0.3s ease-in-out;
+  margin-left: 0.4rem;
+  padding: 0.5rem 1rem;
+}
```
Comment (bdeak4) [inline frontend/src/components/SideBar/SideBarLink.vue:31]: nitpick: add space between css rules

**Review by bdeak4 — DISMISSED** [review]

File: frontend/src/components/SideBar/SideBarLink.vue (hunk 2 above)
Comment (nkelava) [reply frontend/src/components/SideBar/SideBarLink.vue:31]: This will be done in another PR ❤️

File: frontend/src/router/router.js [hunk 3]
```diff
@@ -3,8 +3,8 @@ import { useUserStore } from '@/stores/user';
 
 const routes = [
   {
-    path: '/',
-    name: 'home',
+    path: '/home2',
+    name: 'home2',
```
Comment (abasic) [inline frontend/src/router/router.js:7]: Love the name 😄

File: frontend/src/components/SideBar/SideBarLink.vue [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+.item-text-extended {
+  opacity: 1;
+  color: var(--color-light);
+  transition: opacity 0.8s ease-in-out;
+}
+.item-text-extended:hover {
+  color: var(--color-main);
+}
+.sidebar-item.router-link-exact-active {
+  /* background-color: var(--color-darker); */
```
Comment (abasic) [inline frontend/src/components/SideBar/SideBarLink.vue:66]: Nitpick: I assume this can be removed.

**Review by abasic — APPROVED** [review]

File: frontend/src/router/router.js (hunk 3 above)
Comment (nkelava) [reply frontend/src/router/router.js:7]: Thanks, the naming wizard in me really popped off here 😂 This code will be removed in the future cleanup PR. 😁

File: frontend/src/components/SideBar/SideBarLink.vue (hunk 4 above)
Comment (nkelava) [reply frontend/src/components/SideBar/SideBarLink.vue:66]: Yes, I missed that but will remove it in cleanup PR.

**Comment by abasic** [comment]
@nk38104 Also please resolve the conflicts

**Pull request merged**